aws/gd.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/go-commons/errors"
+)
+
+// getAllGuardDutyDetectors returns active GuardDuty detectors in a given region.
+func getAllGuardDutyDetectors(session *session.Session, excludeAfter time.Time) ([]*string, error) {
+	svc := guardduty.New(session)
+
+	ctx := context.Background()
+	var detectors []*string
+
+	err := svc.ListDetectorsPagesWithContext(ctx, &guardduty.ListDetectorsInput{},
+		func(d *guardduty.ListDetectorsOutput, lastPage bool) bool {
+			for _, detector := range d.DetectorIds {
+				detectors = append(detectors, detector)
+			}
+			return true
+		})
```

Comment:
Don't you need to use the `lastPage` variable here?

Reply:
`lastPage` is part of the method signature of `ListDetectorsPagesWithContext` and is used in that function as part of the paging process. [see here](https://docs.aws.amazon.com/sdk-for-go/api/service/guardduty/#GuardDuty.ListDetectorsPagesWithContext)